crates/vm/levm/src/opcode_handlers/system.rs

```diff
@@ -458,4 +464,143 @@ impl VM {
 
         Ok(OpcodeSuccess::Result(ResultReason::SelfDestruct))
     }
+
+    /// Common behavior for CREATE and CREATE2 opcodes
+    ///
+    /// Could be used for CREATE type transactions
+    // TODO: Improve and test REVERT behavior for CREATE. Issue: https://github.com/lambdaclass/ethrex/issues/1061
+    pub fn create(
+        &mut self,
+        value_in_wei_to_send: U256,
+        code_offset_in_memory: U256,
+        code_size_in_memory: usize,
+        salt: Option<U256>,
+        current_call_frame: &mut CallFrame,
+    ) -> Result<OpcodeSuccess, VMError> {
+        // First: Validations that can cause out of gas.
+        // 1. Cant be called in a static context
+        if current_call_frame.is_static {
+            return Err(VMError::OpcodeNotAllowedInStaticContext);
+        }
+        // 2. Cant exceed init code max size
+        if code_size_in_memory > INIT_CODE_MAX_SIZE {
+            return Err(VMError::OutOfGas(OutOfGasError::ConsumedGasOverflow));
+        }
+
+        // SECOND: Validations that push 0 to the stack
+        let sender_address = current_call_frame.to;
```

Comment:
I am a bit hesitant about "sender_address" variable name; isn't "to" the receiver/creator?

Reply:
Yeah it may be confusing. The thing is that opcodes like create and call send `Messages`. So in this case the sender of this `Create` message is the current executing contract.
If it is still confusing maybe we can rename it to `deployer_address` or something similar

Reply:
|  If it is still confusing maybe we can rename it to deployer_address or something similar

I really like `deployer_contract` 😄 . I think it is clearer.

Reply:
I ended up changing it's name to `deployer_address`, sorry I didn't choose `deployer_contract` but the thing is that I wanted to clarify that it was an address. The name `deployer_contract_address` would've been too long haha.
https://github.com/lambdaclass/ethrex/pull/1460/commits/91a031f4796d40d136605d0dd2d9cf51f7ff920c